browser/mapping.go

```diff
@@ -533,9 +533,17 @@ func mapBrowser(ctx context.Context, vu k6modules.VU, b api.Browser) mapping {
 		"close":       b.Close,
 		"contexts":    b.Contexts,
 		"isConnected": b.IsConnected,
-		"on":          b.On,
-		"userAgent":   b.UserAgent,
-		"version":     b.Version,
+		"on": func(event string) *goja.Promise {
+			return k6ext.Promise(ctx, func() (result any, reason error) {
+				res, err := b.On(event)
+				if err != nil {
+					return nil, err //nolint:wrapcheck
+				}
+				return res, nil
```

Comment:
The following is the same (less code is less things to maintain):

```suggestion
return b.On(event) //nolint:wrapcheck
```